projects/debouncer/src/debouncer.c

```diff
@@ -0,0 +1,58 @@
+#include "debouncer.h"
+
+#include "log.h"
+#include "soft_timer.h"
+#include "stm32f0xx_interrupt.h"
+
+#define INTERRUPT_MASKING_DURATION_MS 50
+
+// This is the callback for the soft timer. Gets the state and compares with old state.
+// If states are equal, it means that we had a button input. So it runs the given callback.
+static void prv_debouncer_timer_callback(SoftTimerID timer_id, void *context) {
+  DebouncerInfo *debouncer_info = context;
+
+  GPIOState current_state;
+  gpio_get_state(&debouncer_info->address, &current_state);
+
+  if (current_state == debouncer_info->state) {
+    debouncer_info->callback(&debouncer_info->address, debouncer_info->context);
```

Comment:
Please add a null check for `debouncer_info->callback`. Although the debouncing would be pretty useless without a callback, it's always good to verify that we aren't trying to call something invalid.